### DISCUSSION TITLE
Braille Pattern font fallback / symbol map

### DISCUSSION THREAD

**Discussion by trevarj** [kovidgoyal/kitty#6152] [General]
I am trying to understand which font Kitty falls back to when displaying [Braille pattern](https://en.wikipedia.org/wiki/Braille_Patterns) characters.

For example, on Kitty when using JetbrainsMono as the default font, the Braille pattern characters render as small squares that are evenly spaced (good). This font doesn't have custom Braille pattern characters, and on other terminal emulators they render as small ovals, which are not evenly spaced.

Kitty w/JetbrainsMono default:
![image](https://user-images.githubusercontent.com/5448324/229434590-afa15f18-91ee-41fe-b09f-08e0ad3c39a9.png)

Alacritty w/JetbrainsMono default: 
![image](https://user-images.githubusercontent.com/5448324/229434696-43e757ad-62bb-47cb-8583-3221e4203658.png)

Does Kitty have a default `symbol_map` for this block of characters?

**Comment by kovidgoyal** [comment]
Characters with defined appearance in the unicode standard are rendered
programitically not from fonts.

On Mon, Apr 03, 2023 at 12:02:34AM -0700, Trevor Arjeski wrote:
> I am trying to understand which font Kitty falls back to when displaying [Braille pattern](https://en.wikipedia.org/wiki/Braille_Patterns) characters.
> 
> For example, on Kitty when using JetbrainsMono as the default font, the Braille pattern characters render as small squares that are evenly spaced (good). This font doesn't have custom Braille pattern characters, and on other terminal emulators they render as small ovals, which are not evenly spaced.
> 
> Kitty w/JetbrainsMono default:
> ![image](https://user-images.githubusercontent.com/5448324/229434590-afa15f18-91ee-41fe-b09f-08e0ad3c39a9.png)
> 
> Alacritty w/JetbrainsMono default: 
> ![image](https://user-images.githubusercontent.com/5448324/229434696-43e757ad-62bb-47cb-8583-3221e4203658.png)
> 
> Does Kitty have a default `symbol_map` for this block of characters?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6152
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by trevarj** [reply]
Ah! Thanks for the clarification.
